src/tools/bit2core.c

```diff
@@ -10,37 +10,42 @@
 
 const int CORE_HEADER_SIZE = 4096;
 const int BITSTREAM_HEADER_SIZE = 120;
+const int BITSTREAM_HEADER_TARGETNAME_LOC = 0x4C;
 
 unsigned char bitstream[MAX_MB*BYTES_IN_MEGABYTE];
 
 typedef struct {
   char name[8];
   int num;
   int max_core_mb_size;
+  char target_device_name[13];
 } rev_info;
 
+static rev_info revs[] =
+{
+  // name   num MB  Target Device Name
+  { "nexys", 1, 4, "7a100tcsg324" },
```

Comment:
Note that in general, if it is 7a100* it will be 4MB, and if it is 7a200* it will be 8MB. But what you have here is fine for now.

Reply:
Yep, had some discussion with rdpeake on this earlier too.

I was thinking of one possible way forward with this would be to permit synonyms here for that 1st field, e.g. something like:

``` { "nexsys4|nexsys4ddr|megaphoner1", 1, 4, "7a100tcsg324" },
```

So for example if the tool was run specifying the **nexys4** argument, but a **nexsys4ddr** bitstream was provided, the tool would still generate a warning, perhaps something like this:

```WARNING: nexys4, exys4ddr and megaphoner1 bitstreams are presently indistinguishable from each other by the tool. Please make your own external confirmations that you have used the correct bitstream for your device.
```

Reply:
In the upcoming commit, I've implemented this 'synonym' mechanism for mega65 target names that use the same FPGA part (E.g., "nexys4|nexsys4ddr|megaphoner1" all use "7a100tcsg324").

So I'll mark this one as resolved.